source/lambda/translatecaptions/translatecaptions.js

```diff
@@ -54,20 +55,15 @@ exports.handler = async (event, context, callback) => {
 
       var captions = JSON.parse(captionsStr);
 
-      for (var i in captions) {
-        var caption = captions[i];
-
-        if (caption.text.trim() === "") {
-          continue;
-        }
-        var captionText = caption.text;
-        var captionTranslatedText = await translateText(
-          sourceLanguage,
-          targetLanguage,
-          captionText
-        );
-        caption.text = escapeHtml(captionTranslatedText);
-      }
+      await transcribePromisePool(50, captions, sourceLanguage, targetLanguage)
```

Comment:
We may set the pool limit to 10, for most user, i think 10 is enough. And later I will change code and CFN template to make it configurable in env configuration